src/services/fs/backend.rs

```diff
@@ -126,6 +126,13 @@ pub struct Backend {
     atomic_write_dir: Option<String>,
 }
 
+#[inline]
+fn tmp_file_of(path: &str) -> String {
+    let name = str::replace(path, "/", "_");
```

Comment:
Seems tricky if user tried to write path `a/b` and `a_b`. How about adding a uuid in the path?

Reply:
Agree, it's tricky here. The `uuid` would be better but may hurt performance. I will change it.

Reply:
> The `uuid` would be better but may hurt performance.

I believe it's not a big problem here:

```rust
group.bench_function("uuid", |b| {
    b.iter(|| {
        let _ = uuid::Uuid::new_v4().to_string();
    })
});
```

```shell
build_abs_path/uuid     time:   [231.20 ns 231.42 ns 231.60 ns]
```

In fact, we can always generate a uuid instead of using origin path.

Reply:
Yep. But I would prefer to append the uuid to the origin file name. It's more readable in debug mode or logging etc.